x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/FullTextFunction.java

```diff
@@ -428,4 +437,20 @@ public static FieldAttribute fieldAsFieldAttribute(Expression field) {
     public void postOptimizationVerification(Failures failures) {
         resolveTypeQuery(query(), sourceText(), forPostOptimizationValidation(query(), failures));
     }
+
+    @Override
+    public BiConsumer<LogicalPlan, Failures> postOptimizationPlanVerification() {
+        // check plan again after predicates are pushed down into subqueries
+        return FullTextFunction::checkFullTextQueryFunctions;
```

Comment:
Is it equivalent to delegate to `postAnalysisPlanVerification()`, so we avoid redefining this method in the subclasses?